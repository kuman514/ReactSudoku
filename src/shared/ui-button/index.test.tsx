import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import { UIButton } from '.';

describe('UIButton', () => {
  beforeEach(() => {
    cleanup();
  });

  it('should have a snapshot match for default', () => {
    const { container } = render(<UIButton>This is a button.</UIButton>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should have a snapshot match for contained', () => {
    const { container } = render(
      <UIButton variant="contained">This is a contained button.</UIButton>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be clickable', async () => {
    const mockFn = vi.fn();
    render(
      <UIButton ariaLabel="test-button" onClick={mockFn}>
        Click Test
      </UIButton>
    );
    fireEvent.click(await screen.findByLabelText('test-button'));
    expect(mockFn).toBeCalledTimes(1);
  });

  it('should be not clickable when it is disabled', async () => {
    const mockFn = vi.fn();
    render(
      <UIButton ariaLabel="test-button" onClick={mockFn} isDisabled>
        Click Test
      </UIButton>
    );
    fireEvent.click(await screen.findByLabelText('test-button'));
    expect(mockFn).toBeCalledTimes(0);
  });

  it('should has customizable style', () => {
    const { container } = render(
      <UIButton
        customStyle={{
          padding: '12px',
          color: '#20cc20',
        }}
      >
        This is a button.
      </UIButton>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
