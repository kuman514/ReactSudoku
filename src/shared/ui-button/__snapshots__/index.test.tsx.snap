// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UIButton > should has customizable style 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-5slo47-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  This is a button.
</button>
`;

exports[`UIButton > should have a snapshot match for contained 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-142372x-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  This is a contained button.
</button>
`;

exports[`UIButton > should have a snapshot match for default 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-b4yaxo-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  This is a button.
</button>
`;
